esphome/components/packet_transport/packet_transport.cpp

```diff
@@ -0,0 +1,461 @@
+    ESP_LOGW(TAG, "Bad length %zu", len);
+    return;
+  }
+  hlen = round4(hlen + 3);
+  buf = start_ptr + hlen;
+  if (buf == end) {
+    ESP_LOGV(TAG, "No data after header");
+    return;
+  }
+
+  if (this->providers_.count(namebuf) == 0) {
+    ESP_LOGVV(TAG, "Unknown hostname %s", namebuf);
+    return;
+  }
+  auto &provider = this->providers_[namebuf];
+  // if encryption not used with this host, ping check is pointless since it would be easily spoofed.
+  if (provider.encryption_key.empty())
+    ping_key_seen = true;
+
+  ESP_LOGV(TAG, "Found hostname %s", namebuf);
+#ifdef USE_SENSOR
+  auto &sensors = this->remote_sensors_[namebuf];
+#endif
+#ifdef USE_BINARY_SENSOR
+  auto &binary_sensors = this->remote_binary_sensors_[namebuf];
+#endif
+
+  if (!provider.encryption_key.empty()) {
+    xxtea::decrypt((uint32_t *) buf, (end - buf) / 4, (uint32_t *) provider.encryption_key.data());
+  }
+  byte = *buf++;
+  if (byte == ROLLING_CODE_KEY) {
+    if (!process_rolling_code(provider, buf, end))
+      return;
+  } else if (byte != DATA_KEY) {
+    ESP_LOGV(TAG, "Expected rolling_key or data_key, got %X", byte);
+    return;
+  }
+  while (buf < end) {
+    byte = *buf++;
+    if (byte == ZERO_FILL_KEY)
+      continue;
+    if (byte == PING_KEY) {
+      if (end - buf < 4) {
+        ESP_LOGV(TAG, "PING_KEY requires 4 more bytes");
+        return;
+      }
+      auto key = get_uint32(buf);
+      if (key == this->ping_key_) {
+        ping_key_seen = true;
+        ESP_LOGV(TAG, "Found good ping key %X", (unsigned) key);
+      } else {
+        ESP_LOGV(TAG, "Unknown ping key %X", (unsigned) key);
+      }
+      continue;
+    }
+    if (!ping_key_seen) {
+      ESP_LOGW(TAG, "Ping key not seen");
+      this->resend_ping_key_ = true;
+      break;
+    }
+    if (byte == BINARY_SENSOR_KEY) {
+      if (end - buf < 3) {
+        ESP_LOGV(TAG, "Binary sensor key requires at least 3 more bytes");
+        return;
+      }
+      rdata.u32 = *buf++;
+    } else if (byte == SENSOR_KEY) {
+      if (end - buf < 6) {
+        ESP_LOGV(TAG, "Sensor key requires at least 6 more bytes");
+        return;
+      }
+      rdata.u32 = get_uint32(buf);
+    } else {
+      ESP_LOGW(TAG, "Unknown key byte %X", byte);
+      return;
+    }
+
+    hlen = *buf++;
+    if (end - buf < hlen) {
+      ESP_LOGV(TAG, "Name length of %u not available", hlen);
+      return;
+    }
+    memset(namebuf, 0, sizeof namebuf);
+    memcpy(namebuf, buf, hlen);
+    ESP_LOGV(TAG, "Found sensor key %d, id %s, data %lX", byte, namebuf, (unsigned long) rdata.u32);
+    buf += hlen;
+#ifdef USE_SENSOR
+    if (byte == SENSOR_KEY && sensors.count(namebuf) != 0)
+      sensors[namebuf]->publish_state(rdata.f32);
+#endif
+#ifdef USE_BINARY_SENSOR
+    if (byte == BINARY_SENSOR_KEY && binary_sensors.count(namebuf) != 0)
+      binary_sensors[namebuf]->publish_state(rdata.u32 != 0);
+#endif
+  }
+}
+
+void PacketTransport::dump_config() {
+  ESP_LOGCONFIG(TAG, "UDP:");
```

Comment:
Probably you meant something like this:

```suggestion
  ESP_LOGCONFIG(TAG, "packet_transport:");
```